usaspending_api/references/v2/views/autocomplete.py

```diff
@@ -204,23 +204,63 @@ def post(self, request):
 class RecipientAutocompleteViewSet(BaseAutocompleteViewSet):
 
     def post(self, request):
-        """Return all recipients matching the provided search text"""
+        """Return all Parents and Recipients matching the provided search text"""
 
         search_text, limit = self.get_request_payload(request)
 
-        queryset = LegalEntity.objects.all()
+        # Return Recipients with valid id entries
+        queryset = LegalEntity.objects.exclude(
+            recipient_unique_id__isnull=True).exclude(
+            recipient_unique_id__exact='')
 
-        queryset = queryset.annotate(similarity=Greatest(TrigramSimilarity('recipient_name', search_text),
-                                                         TrigramSimilarity('recipient_unique_id', search_text))).\
-            order_by('-similarity')
-
-        exact_match_queryset = queryset.filter(similarity=1.0)
-        if exact_match_queryset.count() > 0:
-            queryset = exact_match_queryset
-
-        column_names = ['legal_entity_id', 'recipient_name', 'recipient_unique_id']
-
-        return Response({'results': list(queryset.values(*column_names)[:limit])})
+        # Filter based on search text
+        response = {}
+        # Search and filter for Parent Recipients
+        parents = queryset.annotate(
+            search=SearchVector('recipient_name', 'parent_recipient_unique_id')
+        ).filter(
+            search__icontains=search_text,
+            parent_recipient_unique_id__in=F('recipient_unique_id'))[:limit]
+        # Search and filter for Recipients, excluding Parent Recipients
+        recipients = queryset.annotate(
+            search=SearchVector('recipient_name', 'recipient_unique_id')
+        ).filter(
+            search__icontains=search_text).exclude(
+            parent_recipient_unique_id__in=F('recipient_unique_id'))[:limit]
+
+        # Counts to check for more than 10 total results
+        p_limit = parents.count()
+        r_limit = recipients.count()
+        counts = p_limit + r_limit
+        # If both counts are equal to the limit, distribute results evenly.
+        if p_limit and r_limit == limit:
+            parents = parents[:5]
+            recipients = recipients[:5]
+        # If counts are 10 or less, pass.
+        elif counts < 11:
+            pass
+        # If the total counts are greater than 10, set new limits.
+        elif counts > 10:
+            # If both limits are greater than 5, set limits to 5.
+            if p_limit and r_limit > 5:
```

Comment:
This could be combined with the top-level check to be something like: `if p_limit >= limit and r_limit >= limit:`, since they both result in capping each type to half of `limit`